bind-operator/src/charm.py

```diff
@@ -313,6 +259,46 @@ def _get_remote_relation_data(self) -> dns_record.DNSRecordProviderData | None:
         )
         return relation_data
 
+    def _reconcile(self) -> None:  # noqa: C901
+        """Reconciles."""
+        # Retrieve the current topology of units
+        try:
+            topology = self._topology()
+        except (PeerRelationUnavailableError, PeerRelationNetworkUnavailableError) as err:
+            logger.info("Could not retrieve network topology: %s", err)
+            return
+
+        # If we're the leader and not active, check that the active unit is doing well
+        if self.unit.is_leader() and not topology.is_current_unit_active:
+            self._check_and_may_become_active(topology)
+
+        try:
+            relation_data = self.dns_record.get_remote_relation_data()
+        except ValueError as err:
```

Comment:
Since the exception is caught and the charm is still active, the only way of knowing about this would be monitoring the logs right? Could this potentially mask that the charm is working fine but actually is not updating any records?

Reply:
Taking a look at the [`get_remote_relation_data()`](https://github.com/canonical/dns-operators/blob/reconcile/lib/charms/bind/v0/dns_record.py#L627) function, I actually see that this should not happen and I was overly defensive here.  
The only thing that I could check is a `KeyError` and then I think the response would be only logging (as this would be a controller issue most probably).

Reply:
I changed it to catch an eventual `KeyError` and log it accordingly